pymatviz/ptable/ptable_matplotlib.py

```diff
@@ -512,6 +590,15 @@ def ptable_heatmap_splits(
         kwargs=symbol_kwargs,
     )
 
+    # [Optional] Add annotation
+    if anno_text is not None:
+        projector.add_annotation(
+            text=anno_text or {},
+            pos=anno_pos,
+            text_color=anno_text_color,
+            kwargs=anno_kwargs,
+        )
```

Comment:
how about just passing a single dict to `add_annotation` that includes keys 'text', 'pos', 'color'? otherwise, it can be confusing to remember which things to pass as kwargs and which are separate

Reply:
> it can be confusing to remember which things to pass as kwargs and which are separate

I could relate to this. I think the same holds for other methods of projector as well? I would check this.

https://github.com/janosh/pymatviz/blob/8437a9c228ef56cf4475630aa57863cc1f102fed/pymatviz/ptable/_projector.py#L334-L343

Reply:
I played around with this and realized passing everything as a single dict doesn't seem to work as I expected, perhaps I misunderstood something?

With current method, the design logic is all "global" args is passed as a separate arg (content of the text, position, color) and all others to the `ax.text` (analogy to a child plotter) as `kwargs`. 

If then pack everything into a single `kwargs` dict (or unpack everything with `**kwargs`), then we still need to annotate inside the method docstring what args are expected and such.

Reply:
I must be misunderstanding something, looks like 4d66fe34e494b991382cc2a837022369512a2dd5 doesn't bring much improvement. Because at the end of the day we still need to unpack `kwargs` because the default var name of `plt.text` is not very descriptive inside our ptable plotter (`text(x, y, s, fontdict=None, **kwargs)`) and we still need to map `anno_text` to `s` and such internally.

Perhaps you have a better way around this?

Reply:
maybe `dict.setdefault(key1, dict.pop(key2, default))` can help? example: `kwargs.setdefault('t', kwargs.pop('text', ''))`

Reply:
I think this still doesn't bypass the need to map `t` to `text` inside the `add_annotation` method, also this might make debugging hard as user (or us) wouldn't know what the value really come from (`t` or `text`).

Reply:
Sorry I re-read that code again and apparently I misunderstood. That should map `key2` to `key1`.

Reply:
Sorry I don't think I'm really following, do you mean something like f10bebcb4cdd0d2648ba0a6de0e8508375a7b481? Or if not perhaps can you perhaps push a commit to get me on board?

Reply:
see [365a296](https://github.com/janosh/pymatviz/pull/200/commits/365a29661735f3e9cfa1392a3c3bc1dbecb08b04). if you don't think it's an improvement. feel free to ignore this entire discussion and go back to what you had originally

Reply:
> if you don't think it's an improvement.

This sounds like a harsh comment for me to make. But well...I would prefer personally to revert to what we had originally at this moment, and maybe go back to this if we came up with a better solution (as this is internal and wouldn't impact the public API much), which I don't have right now.

---

With current implementation, pros:
- The interface to the projector method looks cleaner and we don't need to remember what to pass as separate args. (But the IDE would hint and we don't need to remember.)

Cons:
- We still need to handle some keywords separately (`text_color`, `text`) inside the method, as not args are passed to `ax.text` directly.
- We lost the type annotation, and partly the default values (I prefer if we don't need to read the source code to know the default values, otherwise IDE would also show the default values).

Reply:
good points. then let's revert to your earlier implementation

Reply:
This branch should be in the original state if I didn't miss anything, perhaps you could review this again? Thanks!